### PR TITLE
[chore] Update default source_type fixture to include databricks_unity

### PR DESCRIPTION
**Pull request by YS-L**
## Description

This updates the default `source_type` fixture in integration tests to include databricks_unity. Most of the custom source type parametrization are also removed to use this new default in order to have better test coverage for databricks using unity catalog.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

**Review by gee-senbong — APPROVED** [review]

**Comment by YS-L** [comment]
Tests are passing now for databricks, but likely will timeout on databricks_unity. Bumping the timeout to check the runtime required for databricks_unity.

**Comment by YS-L** [comment]
@kchua78 Table / view describe doesn't work yet because of this error:

> FAILED tests/integration/api/test_view_describe.py::test_event_view_describe[databricks_unity] - featurebyte.exception.RecordRetrievalException: [UDF_MAX_COUNT_EXCEEDED] Exceeded query-wide UDF limit of 5 UDFs (limited during public preview). Found 12. The UDFs were: `***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`. SQLSTATE: 54KD0

Can we configure the `spark.databricks.safespark.externalUDF.plan.limit` spark config in the cluster? Reference: https://community.databricks.com/t5/data-engineering/udf-max-count-exceeded-exceeded-query-wide-udf-limit-of-5-udfs/td-p/38120?

File: .github/workflows/test_gen.yaml [hunk 1]
```diff
@@ -333,7 +333,7 @@ jobs:
     needs:
     - build-jar
     runs-on: ubuntu-latest
-    timeout-minutes: 45
+    timeout-minutes: 135
```
Comment (YS-L) [inline .github/workflows/test_gen.yaml:336]: The time required is about 50 to 60 minutes, for example: https://github.com/featurebyte/featurebyte/actions/runs/7322531386/job/19944719602?pr=2013

**Review by gee-senbong — APPROVED** [review]

**Comment by kchua78** [comment]
> @kchua78 Table / view describe doesn't work yet because of this error:
> 
> > FAILED tests/integration/api/test_view_describe.py::test_event_view_describe[databricks_unity] - featurebyte.exception.RecordRetrievalException: [UDF_MAX_COUNT_EXCEEDED] Exceeded query-wide UDF limit of 5 UDFs (limited during public preview). Found 12. The UDFs were: `***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_entropy`,`***`.`featurebyte_github_20231222094939_418165`.`f_count_dict_most_frequent_key_value`. SQLSTATE: 54KD0
> 
> Can we configure the `spark.databricks.safespark.externalUDF.plan.limit` spark config in the cluster? Reference: https://community.databricks.com/t5/data-engineering/udf-max-count-exceeded-exceeded-query-wide-udf-limit-of-5-udfs/td-p/38120?

Updated limit to 25

**Review by kchua78 — APPROVED** [review]

**Comment by YS-L** [comment]
It seems the test can be rather flaky at times. Will merge this PR first and monitor. We should look into making it more stable.

**Pull request merged**